data/spells/scripts/support/findPerson.lua

```diff
@@ -0,0 +1,128 @@
+local LEVEL_LOWER = 1
+local LEVEL_SAME = 2
+local LEVEL_HIGHER = 3
+
+local DISTANCE_BESIDE = 1
+local DISTANCE_CLOSE = 2
+local DISTANCE_FAR = 3
+local DISTANCE_VERYFAR = 4
+
+local directionNames = {
+	[DIRECTION_NORTH] = "north",
+	[DIRECTION_SOUTH] = "south",
+	[DIRECTION_EAST] = "east",
+	[DIRECTION_WEST] = "west",
+	[DIRECTION_NORTHEAST] = "north-east",
+	[DIRECTION_NORTHWEST] = "north-west",
+	[DIRECTION_SOUTHEAST] = "south-east",
+	[DIRECTION_SOUTHWEST] = "south-west"
+}
+
+local levelStrings = {
+	[LEVEL_LOWER] = " is on a lower level to the ",
+	[LEVEL_SAME] = " is to the ",
+	[LEVEL_HIGHER] = " is on a higher level to the "
+}
+
+function onCastSpell(creature, var)
+	if not creature:getPlayer() then
+		return false
+	end
+
+	local targetPlayer = Player(Variant.getString(var))
+	if not targetPlayer then
+		return false
+	end
+
+	if targetPlayer:getGroup():getAccess() and not creature:getGroup():getAccess() then
+		creature:sendCancelMessage(RETURNVALUE_PLAYERWITHTHISNAMEISNOTONLINE)
+		creature:getPosition():sendMagicEffect(CONST_ME_POFF)
+		return false
+	end
+
+	local playerPosition = creature:getPosition()
+	local targetPosition = targetPlayer:getPosition()
+	local distance = {
+		x = playerPosition.x - targetPosition.x,
+		y = playerPosition.y - targetPosition.y,
+		z = playerPosition.z - targetPosition.z
+	}
+
+	local level = LEVEL_SAME
```

Comment:
In this case I find the ternary operator better, because there's really no need for copies.